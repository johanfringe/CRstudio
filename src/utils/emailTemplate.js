// 📁 src/utils/emailTemplate.js
export const emailTemplate = ({ logoSrc, logoAlt, title, intro, ctaUrl, ctaLabel, footer }) => {
  return `
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
      <title>${title}</title>

      <!-- ✅ Inter font (werkt in sommige clients) -->
      <style>
        @import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600&display=swap');
      </style>
    </head>
    <body style="margin: 0; padding: 0; font-family: 'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif; background: white;">
      <div style="max-width: 600px; margin: 40px auto; background: white; border-radius: 8px; padding: 32px;">

        <img src="${logoSrc}" alt="${logoAlt}" style="height: 30px; display: block; margin: 0 auto 48px;" />

        <h1 style="text-align: center; color: #111827; font-size: 20px; font-weight: 600; margin-bottom: 16px;">
          ${title}
        </h1>

        <p style="text-align: center; color: #374151; font-size: 16px; line-height: 1.5; margin-bottom: 32px;">
          ${intro}
        </p>

        <div style="text-align: center; margin-bottom: 32px;">
          <a href="${ctaUrl}" style="
            display: inline-block;
            padding: 10px 48px;
            border-radius: 9999px;
            background-color: #1D4ED8;
            color: white;
            font-weight: 500;
            text-decoration: none;
            font-size: 16px;
            box-shadow: 0 2px 4px rgba(0,0,0,0.1);
          ">
            ${ctaLabel}
          </a>
        </div>

        <p style="text-align: center; color: #6B7280; font-size: 14px;">
          ${footer}
        </p>

      </div>
    </body>
    </html>
  `;
};
