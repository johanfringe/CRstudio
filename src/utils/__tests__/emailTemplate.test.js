// __tests__/emailTemplate.test.js :
import { emailTemplate } from "../emailTemplate";

describe("emailTemplate()", () => {
  const defaultProps = {
    logoSrc: "https://example.com/logo.png",
    logoAlt: "CRstudio",
    title: "Welkom bij CRstudio",
    intro: "Bedankt voor je registratie.",
    ctaUrl: "https://crstudio.online/login",
    ctaLabel: "Inloggen",
    footer: "© 2025 CRstudio",
  };

  test("bevat alle hoofdonderdelen", () => {
    const html = emailTemplate(defaultProps);

    expect(html).toContain("Welkom bij CRstudio");
    expect(html).toContain("Bedankt voor je registratie.");
    expect(html).toContain("https://crstudio.online/login");
    expect(html).toContain("Inloggen");
    expect(html).toContain("© 2025 CRstudio");
    expect(html).toContain(`<img src="${defaultProps.logoSrc}"`);
    expect(html).toContain(`alt="${defaultProps.logoAlt}"`);
  });

  test("bevat <a> tag met correcte href", () => {
    const html = emailTemplate(defaultProps);
    expect(html).toMatch(new RegExp(`<a[^>]+href=["']${defaultProps.ctaUrl}["']`, "i"));
  });

  test("genereert geldige HTML structuur", () => {
    const html = emailTemplate(defaultProps);
    expect(html).toContain("<!DOCTYPE html>");
    expect(html).toContain("<html");
    expect(html).toContain("</html>");
    expect(html).toMatch(/<meta charset=["']UTF-8["']/);
    expect(html).toMatch(/<meta name=["']viewport["']/);
  });

  test("bevat essentiële inline CSS styles", () => {
    const html = emailTemplate(defaultProps);
    expect(html).toMatch(/style="[^"]*font-family: 'Inter'/);
    expect(html).toMatch(/style="[^"]*background-color: #1D4ED8/);
  });

  test("werkt ook met lege strings", () => {
    const html = emailTemplate({
      logoSrc: "",
      logoAlt: "",
      title: "",
      intro: "",
      ctaUrl: "",
      ctaLabel: "",
      footer: "",
    });

    expect(typeof html).toBe("string");
    expect(html).toContain("<html");
    expect(html).toContain("</html>");
  });

  test("genereert consistente output (snapshot)", () => {
    const html = emailTemplate(defaultProps);
    expect(html).toMatchSnapshot();
  });

  test("voorkomt onveilige script injectie in title", () => {
    const html = emailTemplate({ ...defaultProps, title: "<script>alert('x')</script>" });
    expect(html).not.toContain("<script>alert('x')</script>");
    expect(html).toContain("&lt;script&gt;alert(&#039;x&#039;)&lt;/script&gt;");
  });

  test("escape fallback werkt correct bij ontbrekende waarden", () => {
    const html = emailTemplate({
      logoSrc: undefined,
      logoAlt: undefined,
      title: undefined,
      intro: undefined,
      ctaUrl: undefined,
      ctaLabel: undefined,
      footer: undefined,
    });

    expect(typeof html).toBe("string");
    expect(html).toContain("<html");
    expect(html).toContain("</html>");
  });

  test("bevat consistente spacing voor hoofdsecties", () => {
    const html = emailTemplate(defaultProps);
    expect(html).toMatch(/<h1[^>]*>[\s\S]*?<\/h1>/); // inline whitespace validatie
  });

  test("verwerkt speciale karakters correct", () => {
    const html = emailTemplate({ ...defaultProps, title: "Quote \" test & check 'x'" });
    expect(html).toContain("Quote &quot; test &amp; check &#039;x&#039;");
  });
});
