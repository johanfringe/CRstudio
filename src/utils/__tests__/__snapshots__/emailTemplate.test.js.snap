// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emailTemplate() genereert consistente output (snapshot) 1`] = `
"
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
      <title>Welkom bij CRstudio</title>

      <!-- ✅ Inter font (werkt in sommige clients) -->
      <style>
        @import url('https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600&display=swap');
      </style>
    </head>
    <body style="margin: 0; padding: 0; font-family: 'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif; background: white;">
      <div style="max-width: 600px; margin: 40px auto; background: white; border-radius: 8px; padding: 32px;">

        <img src="https://example.com/logo.png" alt="CRstudio" style="height: 30px; display: block; margin: 0 auto 48px;" />

        <h1 style="text-align: center; color: #111827; font-size: 20px; font-weight: 600; margin-bottom: 16px;">
          Welkom bij CRstudio
        </h1>

        <p style="text-align: center; color: #374151; font-size: 16px; line-height: 1.5; margin-bottom: 32px;">
          Bedankt voor je registratie.
        </p>

        <div style="text-align: center; margin-bottom: 32px;">
          <a href="https://crstudio.online/login" style="
            display: inline-block;
            padding: 10px 48px;
            border-radius: 9999px;
            background-color: #1D4ED8;
            color: white;
            font-weight: 500;
            text-decoration: none;
            font-size: 16px;
            box-shadow: 0 2px 4px rgba(0,0,0,0.1);
          ">
            Inloggen
          </a>
        </div>

        <p style="text-align: center; color: #6B7280; font-size: 14px;">
          © 2025 CRstudio
        </p>

      </div>
    </body>
    </html>
  "
`;
